src/foam/nanos/ruler/RulerDAO.js

```diff
@@ -111,7 +111,40 @@ foam.CLASS({
     {
       class: 'Map',
       name: 'rulesList',
-      javaFactory: `return new java.util.HashMap<Predicate, GroupBy>();`
+      synchronized: true,
+      visibility: 'HIDDEN',
+      javaFactory: `
+    ((foam.nanos.logger.Logger) getX().get("logger")).debug("RulerDAO", "initializing rules for", getDaoKey(), getOf().getId());
+    DAO ruleDAO = ((DAO) getX().get("ruleDAO")).where(
+      EQ(Rule.DAO_KEY, getDaoKey())
+    );
+   ruleDAO = ruleDAO.where(
+      EQ(Rule.ENABLED, true)
+    ).orderBy(new Desc(Rule.PRIORITY));
+    ruleDAO.select(new AbstractSink(new ReadOnlyDAOContext(getX())) {
+      @Override
+      public void put(Object obj, Detachable sub) {
+        Rule rule = (Rule) obj;
+        rule.setX(getX());
+      }
+    });
```

Comment:
ruleDAO.listen needs to be between the initial ruleDAO and ruleDAO.where at line 121.

https://github.com/foam-framework/foam2/pull/2166/commits/04f531c28d3437260556c9e22927d56338ebbcb6